server/models/User.ts

```diff
@@ -470,6 +470,20 @@ class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
     }
   };
 
+  /**
+   *  Safely destroy a user account to avoid email conflicts in the future
+   *  Updates the user email before soft-deleting the account
+   */
+  safeDestroy = async function () {
+    // Update user email in order to free up for future reuse
+    // Split the email, username from host domain
+    const splitedEmail = this.email.split('@');
```

Comment:
No need to do anything about it in this case (especially since this comes from older code), but just for sharing the info: the assumption that emails can only contain one `@` and that it separates host from domain is wrong: https://www.netmeister.org/blog/email.html#source-routing